doozer/doozerlib/cli/images_konflux.py

```diff
@@ -188,3 +196,192 @@ async def images_konflux_build(
+        else:
+            # Get latest build records for all specified operators
+            runtime.initialize(mode='images', clone_distgits=False)
+            LOGGER.info("Fetching latest operator builds from Konflux DB...")
+            operator_metas: List[ImageMetadata] = []
+            tasks = []
+            for metadata in runtime.ordered_image_metas():
+                if not metadata.is_olm_operator:
+                    if runtime.images:
+                        LOGGER.warning(f"Skipping non-OLM operator {metadata.distgit_key}")
+                    continue
+                operator_metas.append(metadata)
+                tasks.append(asyncio.create_task(konflux_db.get_latest_build(
+                    name=metadata.distgit_key,
+                    assembly=assembly,
+                    group=self.runtime.group,
+                    el_target=f'el{metadata.branch_el_target()}',
+                    engine=Engine.KONFLUX, strict=True)))
+            records = await asyncio.gather(*tasks, return_exceptions=True)
+            errors = [(meta.distgit_key, record) for meta, record in zip(operator_metas, records) if isinstance(record, Exception)]
+            if errors:
+                raise DoozerFatalError(f"Failed to fetch latest builds from Konflux DB: {errors}")
+            for metadata, record in zip(operator_metas, records):
+                assert record is not None and isinstance(record, KonfluxBuildRecord)
+                dgk_records[metadata.distgit_key] = record
+
+        assert runtime.source_resolver is not None, "source_resolver is not initialized. Doozer bug?"
+        assert runtime.group_config is not None, "group_config is not initialized. Doozer bug?"
+
+        rebaser = KonfluxOlmBundleRebaser(
+            base_dir=Path(runtime.working_dir, constants.WORKING_SUBDIR_KONFLUX_BUILD_SOURCES),
+            group=runtime.group,
+            assembly=assembly,
+            group_config=runtime.group_config,
+            konflux_db=runtime.konflux_db,
+            source_resolver=runtime.source_resolver,
+            upcycle=runtime.upcycle,
+            image_repo=self.image_repo,
+            dry_run=self.dry_run,
+        )
+        db_for_bundles = KonfluxDb()
+        db_for_bundles.bind(KonfluxBundleBuildRecord)
+        builder = KonfluxOlmBundleBuilder(
+            base_dir=Path(runtime.working_dir, constants.WORKING_SUBDIR_KONFLUX_BUILD_SOURCES),
+            group=runtime.group,
+            assembly=assembly,
+            source_resolver=runtime.source_resolver,
+            db=db_for_bundles,
+            konflux_namespace=self.konflux_namespace,
+            konflux_kubeconfig=self.konflux_kubeconfig,
+            konflux_context=self.konflux_context,
+            image_repo=self.image_repo,
+            skip_checks=self.skip_checks,
+            dry_run=self.dry_run,
+        )
+
+        async def _rebase_and_build(image_meta: ImageMetadata, operator_build: KonfluxBuildRecord):
```

Comment:
Is there a strong need for an embedded function? moving it at the class scope would facilitate readability and testing